docs/source/deep_dives/checkpointer.rst

```diff
@@ -404,6 +404,11 @@ looks something like this:
     # set to True if restarting training
     resume_from_checkpoint: True
 
+    # adapter_only option
+    # Set to True to save only the adapter weights for intermediate epochs.
+    # For the final epoch, the entire model weights will be saved regardless of this option.
+    adapter_only: False
```

Comment:
nit: Considering `save_adapter_weights_only` instead of `adapter_only` for clarity. 

I know it's different from the param name on the checkpointer, but the extra context isn't needed there b/c you already know you're dealing with saving a checkpointing.

cc @SalmanMohammadi  and @pbontrager  for thoughts.